hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/impl/HddsDispatcher.java

```diff
@@ -276,7 +277,8 @@ private ContainerCommandResponseProto dispatchRequest(
         getMissingContainerSet().remove(containerID);
       }
     }
-    if (getMissingContainerSet().contains(containerID)) {
+    if (cmdType != Type.CreateContainer && !HddsUtils.isReadOnly(msg)
```

Comment:
Why do we need to expand this check?

Reply:
We should allow writes to go through since we could be creating a RECOVERING container done by replication manager. As of date that is a problem. Once a container is marked missing we don't allow replication manager replicate the same container on the datanode. Consider we have only 5 nodes and we lost one replica because of volume failure now reconstruction will fail forever because the create container command fired would fail forever now.